pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoConsumeSchema.java

```diff
@@ -132,6 +132,11 @@ public void configureSchemaInfo(String topicName,
         }
     }
 
+    @Override
+    public Schema<GenericRecord> cloneSchema() {
+        return Schema.AUTO_CONSUME();
```

Comment:
When you implementing `cloneSchema`, don't you need to copy the fields to the new instance?

Reply:
it don't need to copy fields, AutoConsumeSchema represent it can auto consume, if you want clone the fields you can clone the true schema in AutoConsumeSchema. After all, only its schema is changeable.

Reply:
I understand it can be changed. but `cloneSchema` is cloning a schema, we should copy the fields when cloning a schema.

Reply:
you are right, we should clone the fields when clone a schema